rita_client/src/operator_update/mod.rs

```diff
@@ -335,6 +350,60 @@ fn perform_operator_update(
     trace!("Successfully completed OperatorUpdate");
 }
 
+fn update_authorized_keys(
+    add_list: Vec<String>,
+    drop_list: Vec<String>,
+    keys_file: &str,
+) -> Result<(), std::io::Error> {
+    let mut existing_keys = Vec::new();
+    let mut keys_to_add = Vec::new();
+    let now = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let managed_by = String::from("//managed by OpsTools");
+
+    let key_file = File::open(keys_file);
+    match key_file {
+        Ok(key_file_open) => {
+            let buf_reader = BufReader::new(key_file_open);
+
+            for key in buf_reader.lines() {
+                print!("leaving existing key");
+                existing_keys.push(key.unwrap());
+            }
+            if !add_list.is_empty() {
+                for key in add_list {
+                    if !existing_keys.contains(&key.to_string()) {
+                        keys_to_add.push(key);
+                    }
+                }
+            }
+            if !drop_list.is_empty() {
+                for (pos, key) in drop_list.iter().enumerate() {
+                    if existing_keys.contains(&key.to_string()) {
```

Comment:
I'm not sure contains will work the way you want it to, lets say we try to add key 123abc in the file it will be 123abc // managed by OpsTools 

which won't  match 'contains' (becuase contains on a vec requires equality whereas contains on a string is a substring match). It may help to iterate over and do contains on each string within the vec using a utility function. These files are small so I wouldn't worry about iterating over them several times during this process.